crates/tx-parser/src/filter.rs

```diff
@@ -1,92 +1,94 @@
-///  info
-fn check_and_extract_relevant_info(
+///  info.
+///  TODO: make this function return multiple ops as a single tx can have multiple outpoints that's
+///  relevant
+fn extract_protocol_op(
     tx: &Transaction,
-    filters: &[TxFilterRule],
+    filter_conf: &TxFilterConfig,
 ) -> Option<ProtocolOperation> {
-    filters.iter().find_map(|rel_type| match rel_type {
-        TxFilterRule::RollupInscription(name) => {
-            if !tx.input.is_empty() {
-                if let Some(scr) = tx.input[0].witness.tapscript() {
-                    if let Ok(inscription_data) = parse_inscription_data(&scr.into(), name) {
-                        if let Ok(signed_batch) = borsh::from_slice::<SignedBatchCheckpoint>(
-                            inscription_data.batch_data(),
-                        ) {
-                            return Some(ProtocolOperation::RollupInscription(signed_batch));
-                        }
-                    }
-                }
-            }
-            None
-        }
+    // Currently all we have are inscription txs, deposits and deposit requests
+    parse_inscription_checkpoint(tx, filter_conf)
+        .map(ProtocolOperation::Checkpoint)
+        .or_else(|| parse_deposit(tx, filter_conf).map(ProtocolOperation::Deposit))
+        .or_else(|| parse_deposit_request(tx, filter_conf).map(ProtocolOperation::DepositRequest))
+}
 
-        TxFilterRule::DepositRequest(config) => extract_deposit_request_info(tx, config)
-            .map(|deposit_req_info| Some(ProtocolOperation::DepositRequest(deposit_req_info)))?,
+/// Identifies the tx utxos that are spent to certain addresses/script_pubkeys and parse any
+/// information accordingly.
+fn _parse_spent_to(tx: &Transaction, filter_conf: &TxFilterConfig) -> Option<ProtocolOperation> {
+    // Check for spent to addrs
+    let _matched_op = tx
+        .output
+        .iter()
+        .filter_map(|op| {
+            filter_conf
+                .expected_addrs
+                .as_slice()
+                .binary_search_by_key(&op.script_pubkey, |e| e.address().script_pubkey())
+                .ok()
+                .and_then(|idx| filter_conf.expected_addrs.get(idx))
+        })
+        .next();
+    // TODO: complete this when we figure out what kind of protocol ops can be generated from
+    // matching spends
+
+    None
+}
 
-        TxFilterRule::Deposit(config) => extract_deposit_info(tx, config)
-            .map(|deposit_info| Some(ProtocolOperation::Deposit(deposit_info)))?,
+fn parse_deposit_request(
+    tx: &Transaction,
+    filter_conf: &TxFilterConfig,
+) -> Option<DepositRequestInfo> {
+    extract_deposit_request_info(tx, &filter_conf.deposit_config)
+}
+
+fn parse_deposit(tx: &Transaction, filter_conf: &TxFilterConfig) -> Option<DepositInfo> {
+    extract_deposit_info(tx, &filter_conf.deposit_config)
+}
+
+/// Parses inscription from the given transaction. The inscription currently recognized is the
+/// checkpoint inscription.
```

Comment:
```suggestion
/// Parses inscription from the given transaction. Currently, the only inscription recognizable is the
/// checkpoint inscription.
```
I think this is less ambiguous and captures your intentions more clearly, no?